### ISSUE TITLE
Allow createInstance logger's to be scoped to the instance rather than global

### ISSUE THREAD

**Issue by jrbarnard** [optimizely/javascript-sdk#615]
##  How would the enhancement work?

Currently the createInstance function sets up an Optimizely instance with a global logger. This works fine when creating a single instance, but if for some reason you create two, the subsequent call to createInstance overwrites the global log handler being used by both instances. This means any prefixes / log levels etc that were set are then changed across both instances.

It would be ideal if there were a way for log handlers to be scoped to the instance being created.

## When would the enhancement be useful?

When creating more than 1 optimizely instance.

Let me know if you need any further detail, thanks

**Comment by mikechu-optimizely** [comment]
Hi @jrbarnard,

Is this still something you need reviewed?

I see what you mean where, if you use `createInstance` there's an implied use of the [`globalLogManager`](https://github.com/optimizely/javascript-sdk/blob/master/lib/modules/logging/logger.ts#L299C5-L299C21) ... which as you pointed out would act like a static single point log handler throughout all instances of the Opti client.  🤔   I think this was intentional to give the fastest on ramp to experimentation, where a given app would likely only run one instance of the Opti client in the app.

A less fun workaround might be to instantiate the SDK using the [primary `constructor`](https://github.com/optimizely/javascript-sdk/blob/master/lib/optimizely/index.ts#L100) perhaps giving you the opportunity to pass by ref a logger (your own custom or an instance of the [default logger](https://github.com/optimizely/javascript-sdk/blob/master/lib/modules/logging/logger.ts#L207) with a `messagePrefix`).  

Each `new Optimizely()` instances would then have a separate logger, right?

I might be way off here, so please let me know your thoughts when you have a chance.

**Comment by mikechu-optimizely** [comment]
Yeah, 😬 We've not shown enough love ❤️  to GitHub Issues....which I'm trying to fix. 

Thanks for the pingback though.

**Comment by mikechu-optimizely** [comment]
Actually, this may be a helpful concept.  I've opened an internal ticket (FSSDK-9616) to cover, but we'll keep this Issue closed as to not bombard you. 😁
